src/components/orderDetails/OrderDetails.tsx

```diff
@@ -0,0 +1,345 @@
+  EMAIL,
+  ITEMS_INFO,
+  LOCALE,
+  NAME,
+  NO_MATCHING_ORDERS_FOUND,
+  OMS,
+  ORDER,
+  ORDER_DETAILS,
+  ORDER_TOTAL,
+  ORDER_TYPE,
+  PAYMENT_INFO,
+  PAYMENT_TYPE,
+  PROMOTIONS,
+  STATUS,
+  STATUS_ACROSS,
+  SUBMITTED,
+  VIEW_ALL,
+  WMS,
+} from "../../constants/appConstants";
+import {
+  CommerceItem,
+  CommerceItemData,
+  OmsOrderStatus,
+  OrderData,
+} from "../../@types/OrderDetails";
+import { fetchData } from "../../utils/fetchUtil";
+import Table from "../common/Table";
+import CustomIcon from "../common/CustomIcon";
+import Card from "../common/Card";
+import PromotionsIcon from "../../assets/promotions_white.svg";
+import OrderClockIcon from "../../assets/order_clock_white.svg";
+import OmsInfoIcon from "../../assets/oms_info_white.svg";
+import Loader from "../loader";
+
+const OrderDetails: React.FC = () => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [orderData, setOrderData] = useState<OrderData | Record<string, never>>(
+    {},
+  );
+  const [omsOrderStatus, setOmsOrderStatus] = useState<
+    OmsOrderStatus | Record<string, never>
+  >({});
+  const [omsOrderFlow, setOmsOrderFlow] = useState<Record<string, never>>({});
+  const [promotions, setPromotions] = useState<Record<string, never>>({});
+  const [itemTableData, setItemTableData] = useState<CommerceItemData[]>([]);
+
+  const { orderId } = useParams<{ orderId: string }>();
+
+  useEffect(() => {
+    setIsLoading(true);
+    getOrderData();
+    getOmsOrderStatus();
+    getOmsOrderFlow();
+    getPromotions();
+  }, [orderId]);
+
+  const getOrderData = async () => {
+    const data: OrderData = await fetchData(
+      `${URL_ORDER_DETAILS}/${orderId}`,
+      {},
+    );
+    setOrderData(data || {});
+    if (data?.commerceItem?.length) {
+      const convertedArray: CommerceItemData[] = data.commerceItem.map(
+        (item: CommerceItem) => {
+          return {
+            SKU: item.catalogRefId,
+            Type: item.itemClassType,
+            Quantity: item.quantity,
+            "Unit Price": item.priceInfo.listPrice,
+            "Total Price": item.priceInfo.rawTotalPrice,
+          };
+        },
+      );
+      setItemTableData(convertedArray);
+    }
+    setIsLoading(false);
+  };
+
+  const getOmsOrderStatus = async () => {
+    const data: OmsOrderStatus = await fetchData(
+      `${URL_OMS_ORDER_STATUS}/${orderId}`,
+      {},
+    );
+    setOmsOrderStatus(data || {});
+  };
+
+  const getOmsOrderFlow = async () => {
+    const data: OmsOrderFlow = await fetchData(
+      `${URL_OMS_ORDER_FLOW}/${orderId}`,
+      {},
+    );
+    console.log(omsOrderFlow);
+    setOmsOrderFlow(data || {});
+  };
+
+  const getPromotions = async () => {
+    const promoUrl = URL_PROMOTIONS.replace(":orderId", orderId);
+    const data: OmsOrderFlow = await fetchData(promoUrl, {});
+    console.log(promotions);
+    setPromotions(data || {});
+  };
+
+  const formatName = (fullName): string => {
+    return fullName
```

Comment:
This approachs works. But there is an even simpler approach (text-transform: capitalize)
https://tailwindcss.com/docs/text-transform#transforming-text

Reply:
The first and last names of values from the API response come in Capital letters. So the capitalize class is not working as expected here. That's why used the function to format names.